File: src/main/java/cc/ifnot/demo/controller/ApiController.java
package cc.ifnot.demo.controller;

import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.HashMap;

@RestController
@RequestMapping("/api")
public class ApiController {

    @RequestMapping(value = "/get", method = RequestMethod.POST)
    public HashMap<String, Object> get(@RequestParam String name) {
        HashMap<String, Object> map = new HashMap<>();
        map.put("title", "hello world");
        map.put("name", name);
        return map;
    }
}

File: src/main/java/cc/ifnot/demo/controller/ApiController.kt
package cc.ifnot.demo.controller

import org.springframework.web.bind.annotation.*

import java.util.HashMap

@RestController
@RequestMapping("/api")
class ApiController {

    @CrossOrigin(origins = arrayOf("http://localhost:8080"))
    @RequestMapping(value = "/get", method = arrayOf(RequestMethod.POST))
    operator fun get(@RequestParam name: String): HashMap<String, Any> {
        val map = HashMap<String, Any>()
        map["title"] = "hello world"
        map["name"] = name
        return map
    }
}
